types/validator_set.go

```diff
@@ -711,27 +710,97 @@ func (vals *ValidatorSet) VerifyFutureCommit(newSet *ValidatorSet, chainID strin
 		// }
 	}
 
-	if oldVotingPower <= oldVals.TotalVotingPower()*2/3 {
-		return errTooMuchChange{oldVotingPower, oldVals.TotalVotingPower()*2/3 + 1}
+	if got, needed := float32(oldVotingPower), float32(oldVals.TotalVotingPower()*2/3); got <= needed {
+		return ErrTooMuchChange{Got: got, Needed: needed}
+	}
+	return nil
+}
+
+// VerifyCommitTrusting verifies that trustLevel ([1/3, 1]) of the validator
+// set signed this commit.
+// NOTE the given validators do not necessarily correspond to the validator set
+// for this commit, but there may be some intersection.
+func (vals *ValidatorSet) VerifyCommitTrusting(chainID string, blockID BlockID,
```

Comment:
Can we combine this and the above function? They seem to duplicate a lot of code.

Reply:
In a separate PR. I am not going to touch old lite client in this PR.

Reply:
OK.